packages/suite/src/components/firmware/ReconnectDevicePrompt.tsx

```diff
@@ -226,23 +227,28 @@ export const ReconnectDevicePrompt = ({ onClose, onSuccess }: ReconnectDevicePro
     };
 
     const getSecondStep = () => {
-        const deviceFwVersion = getFirmwareVersion(uiEvent?.payload.device);
-        const deviceModelInternal = uiEvent?.payload.device.features?.internal_model;
-
         if (toNormal) {
             return 'FIRMWARE_CONNECT_IN_NORMAL_MODEL_NO_BUTTON';
         }
 
-        return pickByDeviceModel(deviceModelInternal, {
-            default: 'TR_SWITCH_TO_BOOTLOADER_HOLD_LEFT_BUTTON',
-            [DeviceModelInternal.T1B1]:
-                semver.valid(deviceFwVersion) && semver.satisfies(deviceFwVersion, '<1.8.0')
-                    ? 'TR_SWITCH_TO_BOOTLOADER_HOLD_BOTH_BUTTONS'
-                    : 'TR_SWITCH_TO_BOOTLOADER_HOLD_LEFT_BUTTON',
-            [DeviceModelInternal.T2T1]: 'TR_SWITCH_TO_BOOTLOADER_SWIPE_YOUR_FINGERS',
-            [DeviceModelInternal.T2B1]: 'TR_SWITCH_TO_BOOTLOADER_HOLD_LEFT_BUTTON',
-            [DeviceModelInternal.T3T1]: 'TR_SWITCH_TO_BOOTLOADER_SWIPE_YOUR_FINGERS',
-        } as const);
+        if (deviceModelInternal !== undefined) {
+            const deviceFwVersion = getFirmwareVersion(uiEvent?.payload.device);
+            const switchToBootloaderMap: Record<DeviceModelInternal, TranslationKey> = {
+                [DeviceModelInternal.T1B1]:
+                    semver.valid(deviceFwVersion) && semver.satisfies(deviceFwVersion, '<1.8.0')
+                        ? 'TR_SWITCH_TO_BOOTLOADER_HOLD_BOTH_BUTTONS'
+                        : 'TR_SWITCH_TO_BOOTLOADER_HOLD_LEFT_BUTTON',
+                [DeviceModelInternal.T2T1]: 'TR_SWITCH_TO_BOOTLOADER_SWIPE_YOUR_FINGERS',
+                [DeviceModelInternal.T2B1]: 'TR_SWITCH_TO_BOOTLOADER_HOLD_LEFT_BUTTON',
+                [DeviceModelInternal.T3B1]: 'TR_SWITCH_TO_BOOTLOADER_HOLD_LEFT_BUTTON',
+                [DeviceModelInternal.T3T1]: 'TR_SWITCH_TO_BOOTLOADER_SWIPE_YOUR_FINGERS',
+            };
+
+            return switchToBootloaderMap[deviceModelInternal];
+        }
+
+        // Fallback. This should never happen.
+        return 'TR_SWITCH_TO_BOOTLOADER_HOLD_LEFT_BUTTON';
```

Comment:
Nit: I'd rather do early return to handle the `undefined` case. But that's just my opinion, I am not sure if we have a convention to prefer early returns.

Reply:
06cd5504ff7681bebc9e5da5ebee1db05c068222 Yes, probably that's better. I also fixed a bug I created by the previous fixup and added some explanation to why it is done this way.